fix: remove darwin support for nix

pcnc opened the pull request:
https://supabase.slack.com/archives/C07D7H645L0/p1723346029686629

pcnc committed:
fix: remove darwin support for nix

samrose reviewed (approved):
seems worth a try thank you

samrose commented:
I have a fix I am going to apply to this PR by pushing it right to the branch coming up in a moment

samrose committed:
fix: cut down the size of nix store prior to release of AMI
by running nix-collect-garbage -d which should remove items
not in use by installed packages
